### PR TITLE
Support chat using Zendesk Messaging - attempt 2

### PR DESCRIPTION
**Pull request by klimeryk**
Reverts Automattic/wp-calypso#77016

Second attempt at https://github.com/Automattic/wp-calypso/pull/76777

We're switching from Happy Chat to Zendesk Messaging for live chat support. This PR is big, but it's also the smallest I could get given the scope of the project. Any cleanup of Happy Chat code, old contact forms, etc. will be done as a follow-up. I want to keep this PR as easy to revert as possible if needed.

## Proposed Changes

* Disable Happy Chat feature flag and any availability checks that trigger connections to the websocket.
* Load the Zendesk Messaging script instead.
* Authenticate using already established endpoint (just with `type=zendesk`).
* Check availability using the `/help/messaging/is-available` endpoint.
* Pass the initial chat data using Zendesk user fields.
* Use the support history to figure out if the user has ongoing conversations or not (thankfully, a Messaging conversation is a "ticket"!).
* Additionally, make sure the Zendesk launcher is visible if any unread messages are reported by Zendesk's APIs directly. This is a good safeguard and makes it more dynamic.
* After a refresh, if there's an active Messaging conversation, the Zendesk launcher should automatically appear.
* If the browser is blocking requests to Zendesk, show a notice and link to relevant support doc.
* **New**: support activity is fetched from data stored on the WPCOM user. No external APIs (like Zendesk's) are contacted. That will allow us to provide the best UX, without running into throttling issues.

## Testing Instructions

The live chat support flow should feel the same up to the point where you click the chat CTA (`Chat with us`/`Still chat with us`). At that point, the Help Center should get hidden, while the Zendesk launch and widget should become visible.

Screenshot 2023-05-16 at 19 05 25

Trying to show the Help Center should automatically minimize the Zendesk widget, to avoid clutter and confusing UI. The Zendesk launcher remains visible, so the user can come back to the conversation at any time.

If you have Enhanced Tracking Protection (or similar strict privacy settings and/or ad-blockers), you will see this notice when choosing the Live Chat support option:
Screenshot 2023-05-16 at 19 07 32

Other support options should not show this notice, of course.

Verify that the Zendesk widget follows you on different paths in Calypso. Where Help Center is, the widget should also be there.

If you have an active, ongoing conversation, it should be detected and on refreshes the launcher should appear. If new messages arrive, it will be reflected with a count/notification on the launcher as well.

Screenshot 2023-05-16 at 17 41 35

Verify these behaviors in different environments that Help Center has to work in:
 - Simple sites
 - Atomic sites
 - Editor

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php [hunk 1]
```diff
@@ -88,6 +108,27 @@ public function get_all_support_eligibility() {
...
+	 * @param \WP_REST_Request $request    The request sent to the API.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function get_messaging_support_eligibility( \WP_REST_Request $request ) {
+		$query_parameters = array(
+			'group'       => $request['group'],
+			'environment' => $request['environment'],
+		);
+		$body             = Client::wpcom_json_api_request_as_user( 'help/messaging/is-available?' . http_build_query( $query_parameters ) );
```
Comment (arcangelini) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php:123]:
```suggestion
		$body = Client::wpcom_json_api_request_as_user( '/help/messaging/is-available?' . http_build_query( $query_parameters ) );
```

I could be wrong but I think you need the starting slash here.

Comment (arcangelini) [reply apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php:123]: I guess not, I can see an instance of it without. Forgive me : )

Comment (klimeryk) [reply apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php:123]: I'd assume so too, but looking at the other calls in this file and some others, it seems both versions work? 🤔

Comment (klimeryk) [reply apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php:123]: With a slash: https://github.com/Automattic/wp-calypso/blob/6f2928a02c45d0a52fcf96081ad13ec647431f05/apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-ticket.php#L78
Without: https://github.com/Automattic/wp-calypso/blob/6f2928a02c45d0a52fcf96081ad13ec647431f05/apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-authenticate.php#L44

Those are existing calls that I assume are working. We could definitely update them to be consistent (probably the starting slash is the best way to go).

Comment (klimeryk) [reply apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-support-availability.php:123]: No worries, I appreciate the attention to detail! Those missing starting slashes in most cases _are_ problematic. So probably good to fix this in a follow-up patch.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7996269

Thank you @klimeryk for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7996269).

**Pull request merged**